extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java

```diff
@@ -186,6 +197,10 @@ void build(CombinedIndexBuildItem index,
                 }
             }
         }
+
+        panacheEntities.addAll(modelClasses);
+
+        recordPanacheEntityPersistenceUnits(recorder, jpaModelPersistenceUnitMapping, panacheEntities);
```

Comment:
Hmmm. Do we support using both approaches correctly?

I haven't checked thoroughly but my bet would be it won't work.

Reply:
I'm not sure I was clear but I meant: mixing Panache entities and Panache repositories approaches.

Reply:
I am not sure it is clear enough. You mean a `PanacheEntity` cannot have the corresponding Panache repository? 
If so, I think it is very possible. I once used that pattern here: 
- [TodoRepository.java](https://github.com/machi1990/todo-api-with-quarkus-oidc/blob/master/machi-todo-api/src/main/java/org/machi/api/repository/TodoRepository.java) 
- [Todo.java](https://github.com/machi1990/todo-api-with-quarkus-oidc/blob/master/machi-todo-api/src/main/java/org/machi/api/entity/Todo.java)

Reply:
No I mean: what happens if you have:
- Entity1 which is an entity using the Active Record pattern and inheriting from PanacheEntity
- Entity2 which uses the Panache repository pattern
all in the same project.

Reply:
This should work too. 

It's what I've in my test case I just did not extend it enough: let me add it so that we can confirm that it works.

Reply:
@gsmet I've added this test case in a separate commit https://github.com/quarkusio/quarkus/pull/11852/commits/789e6ed9d0b20461d3258c3357e107b401353e1c

Reply:
OK, perfect if we have a test!